buildroot-generator.py

```diff
@@ -195,7 +210,8 @@ def dump_data(path, data):
         query = base.sack.query().filterm(pkg=base.transaction.install_set)
 
         # Save package dependency data
-        binary_pkg_relations = _analyze_package_relations(query)
+        new_binary_pkg_relations = _analyze_package_relations(query)
+        _update_package_relations_dict(new_binary_pkg_relations, binary_pkg_relations)
```

Comment:
I'm not familiar with the overall operation of this code, but is it important to _not_ overwrite old entries in `binary_pkg_relations` when merging with the new relations? If that's the case, this looks good to me! Otherwise, you should be able implement this as just `binary_pkg_relations.update(_analyze_package_relations(query))`.

Reply:
Yes, exactly.

I might be adding some additional logic to that function, that's why it's separate.

Reply:
(it's not just a flat dict merge)